src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        addLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            addLast(value);
+        } else {
+            addInside(value, findNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T replacedItem = findNode(index).item;
+        findNode(index).item = value;
+        return replacedItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedItem = findNode(index).item;
+        unlink(findNode(index));
+        return removedItem;
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = first; x != null; x = x.next) {
+            if ((object != null && object.equals(x.item)) || object == x.item) {
+                unlink(x);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addLast(T value) {
+        final Node<T> lastElement = last;
+        final Node<T> newNode = new Node<>(lastElement, value, null);
+        last = newNode;
+        if (lastElement == null) {
+            first = newNode;
+        } else {
+            lastElement.next = newNode;
+        }
+        size++;
+    }
+
+    private void chekIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Element with index " + index + " not exist");
+        }
+    }
+
+    private Node<T> findNode(int index) {
+        chekIndex(index);
+        Node<T> node = (index >= (size / 2) ? last : first);
+        int middleIndex = (index >= (size / 2) ? size - index - 1 : index);
+        for (int i = 0; i < middleIndex; i++) {
+            node = (index >= (size / 2) ? node.prev : node.next);
```

Comment:
you use <(index >= (size / 2)> 3 times in this method, maybe you can create a local variable for this?